docs/content/any/reference/polar/polar-syntax.md

```diff
@@ -586,3 +605,290 @@ To add an inline query to a Polar file, use the `?=` operator:
 ```
 
 An inline query is only valid at the beginning of a line.
+
+### Rule Types
+
+A rule type specifies the _shape_ of a rule — its arity and, optionally, the type of each of its arguments. If a rule type exists for `has_permission()`, then all `has_permission()` rules must conform to the rule type.
```

Comment:
since you didn't want to use 'arity' in user-facing docs

```suggestion
A rule type specifies the _shape_ of a rule — the number of arguments it has and, optionally, the type of each argument. If a rule type exists for `has_permission()`, then all `has_permission()` rules must conform to the rule type.
```